public/app/features/dashboard-scene/pages/DashboardScenePageStateManager.ts

```diff
@@ -182,10 +182,6 @@ export class DashboardScenePageStateManager extends StateManagerBase<DashboardSc
         restoreDashboardStateFromLocalStorage(dashboard);
       }
 
-      if (!(config.publicDashboardAccessToken && dashboard.state.controls?.state.hideTimeControls)) {
-        dashboard.startUrlSync();
-      }
```

Comment:
Wonder what this was for? @juanicabanas

Reply:
ah i see, the lack of `UrlSyncContextProvider` in `PublicDashboardScenePage` is actually responsible for this

Reply:
this is basically for not allowing a public dashboard viewer to manage the time range from the URL if the public dashboard configuration has `Time range picker enabled` as false.
I'm not aware of `UrlSyncContextProvider`. Isn't it part of `PublicDashboardScenePage`. If that manages the URL sync, we should include it for the scenario `Time range picker enabled` is true, right?

Reply:
@juanicabanas feels like time range restrictions should be handled in the backend query handling, disabling URL sync won't prevent someone from issuing a query with a different time range

Reply:
yes, we're already handling this restriction in the backend [here](https://github.com/grafana/grafana/blob/2403665998cb91757c826d51f2586168b0db40e2/pkg/services/publicdashboards/validation/validation.go#L19).
It doesn't make sense to add this validation in the frontend because, even though the user could see a URL change when changing the time picker, we're not even rendering the time picker in the pubdash when the config disables it.
My bad here!

Reply:
Wrapping the public dashboard scene in UrlSyncContextProvider only when hideTimeControls is falsy now. 

but feels like the wrong way to enforce time restrictions

Reply:
I saw the change. If you discard that condition and always render the pubdash with UrlSyncContextProvider, the behavior would be the same thanks to the backend validation. But I think it's fine not to render things that should not be there

Reply:
@juanicabanas but this is not rendering anything? it's just initializing a URL sync system that has nothing to do with hiding time controls, without it many dashboard features that we want even in public dashboards would stop working